packages/teamscale-coverage-collector/src/main.ts

```
@@ -281,6 +229,65 @@ export class Main {
 		}
 	}
+
+	private static async uploadToToTeamscale(config: Parameters, logger: Logger, coverageFile: string, lines: number) {
+		if (!(config.teamscale_access_token && config.teamscale_user && config.teamscale_server_url)) {
+			logger.error('Cannot upload to Teamscale: API key and user name must be configured!');
+			return;
+		}
+
+		if (lines === 0) {
+			return;
+		}
+
+		logger.info('Preparing upload to Teamscale');
+
+		const form = new FormData();
+		form.append('report', fs.createReadStream(coverageFile), 'coverage.simple');
+
+		const parameters = new QueryParameters();
+		parameters.addIfDefined('format', 'SIMPLE');
+		parameters.addIfDefined('message', config.teamscale_message);
+		parameters.addIfDefined('repository', config.teamscale_repository);
+		parameters.addIfDefined('t', config.teamscale_commit);
+		parameters.addIfDefined('revision', config.teamscale_revision);
+		parameters.addIfDefined('partition', config.teamscale_partition);
+
+		await axios
+			.post(
+				`${config.teamscale_server_url.replace(/\/$/, '')}/api/projects/${
+					config.teamscale_project
+				}/external-analysis/session/auto-create/report?${parameters.toString()}`,
+				form,
+				{
+					auth: {
+						username: config.teamscale_user,
+						password: config.teamscale_access_token
+					},
+					headers: {
+						Accept: '*/*',
+						'Content-Type': `multipart/form-data; boundary=${form.getBoundary()}`
+					}
+				}
+			)
+			.catch(function (error) {
+				if (error.response) {
+					const response = error.response;
+					if (response.status >= 400) {
+						logger.error(`Upload failed with code ${response.status}: ${response.statusText}`);
+						logger.error(`Request failed ${response.status} with following response: ${response.data}`);
+					} else {
+						logger.info(`Upload with status code ${response.status} finished.`);
+					}
+				} else if (error.request) {
+					logger.error(`Upload request did not receive a response.`);
+				}
+
+				if (error.message) {
```

now if error.response == error.request == error.message == null we log nothing. Please add an else clause and just log the entire error object by converting it to JSON as a last resort fallback so we are 100% certain we don't swallow any errors by accident.